test/unit/StoryChunkTest.cpp

```diff
@@ -0,0 +1,1117 @@
+//     }
+
+//     // Now we insert one more event which should get rejected
+//     chl::LogEvent extraEvent(storyId, startTime + chunkCapacity, 0, 0, "extraevent");
+//     EXPECT_EQ(chunk.insertEvent(extraEvent), 0);
+
+//     // the total event count should be same as the capacity
+//     EXPECT_EQ(chunk.getEventCount(), chunkCapacity);
+// }
+
+// Insert a very large event record
+TEST(StoryChunk_TestInsertEvent, testInsertLargeEventRecord)
+{
+    int storyId(2);
+    int startTime = 0, endTime = 1000;
+    chl::StoryChunk chunk("ChronicleName", "StoryName", storyId, startTime, endTime, 1);
+
+    std::string largePayload(20 * 1024, 'A');
+    chl::LogEvent largeEv(storyId, startTime + 1, 42, 7, largePayload);
+
+    EXPECT_EQ(chunk.insertEvent(largeEv), 1);
+    EXPECT_EQ(chunk.getEventCount(), 1);
+
+    std::vector<chl::Event> series;
+    chunk.extractEventSeries(series);
+    ASSERT_EVENTS_ORDERED(series);
+    ASSERT_EQ(series.size(), 1);
+
+    EXPECT_EQ(series[0].log_record().size(), largePayload.size());
+    EXPECT_EQ(series[0].log_record(), largePayload);
+    EXPECT_TRUE(chunk.empty());
+}
+
+/* ----------------------------------
+  Tests on mergeEvents()
+  ---------------------------------- */
+
+// Need to init chrono logger to avoid abort when we call mergeevents()
+static void initLogger()
+{
+    int ret = chl::chrono_monitor::initialize("console", "", spdlog::level::debug, "unit_test_logger");
+    ASSERT_EQ(ret, 0);
+}
+
+TEST(StoryChunk_TestMergeEvents, testEmptyInputMap)
+{
+    initLogger();
+
+    std::map<chl::EventSequence, chl::LogEvent> events;
+    auto mergeStart = events.cbegin();
+    chl::StoryChunk chunk("ChronicleName", "StoryName", 1, 100, 200, 10);
+
+    // try to merge empty event
+    int retMerge = chunk.mergeEvents(events, mergeStart);
+    EXPECT_EQ(retMerge, 0);
+    EXPECT_TRUE(events.empty());
+    EXPECT_EQ(chunk.getEventCount(), 0);
+}
+
+// Merge events should only merge the events within the current story chunks start and end widnow
+// So we create a set of events, one of them has a time before the shunks start time
+// should get rejected, shouldnt be merged
+TEST(StoryChunk_TestMergeEvents, testStartBeforeWindow)
+{
+    initLogger();
+
+    std::map<chl::EventSequence, chl::LogEvent> events;
+    int startTime = 100, endTime = 200;
+    int storyId(1);
+    events.insert({{50, 0, 0}, chl::LogEvent(storyId, 50, 0, 0, "eventbeforestart")});
+    events.insert({{120, 1, 1}, chl::LogEvent(storyId, 120, 1, 1, "eventafterstart")});
+    events.insert({{220, 0, 0}, chl::LogEvent(storyId, 50, 0, 0, "eventafterstart")});
+
+    auto mergeStart = events.cbegin();
+    chl::StoryChunk chunk("ChronicleName", "StoryName", storyId, startTime, endTime, 10);
+    int retMerge = chunk.mergeEvents(events, mergeStart);
+    EXPECT_EQ(retMerge, 1);
+    EXPECT_EQ(chunk.getEventCount(), 1);
+    ASSERT_EQ(events.size(), 2);
+    EXPECT_EQ(events.begin()->second.time(), 50);
```

Comment:
`{50, 0, 0}` and `{220, 0, 0}` are out of the window `[100, 200]`. should this be 120 instead of 50?

Reply:
I looked at the time of `EventSequence` as the key to insert. Then we all agreed in the meeting that the time of the `LogEvent` should be the one to be checked with the assumption that the `EventSequence` has a matching time. So this is not an issue anymore.